### PR TITLE
Zarmada/working/adts 155

### PR DESCRIPTION
**Pull request by JoseZarmada**
### Testing 🧪
 > [ Add any special instructions needed to test or view your changes such as environment URLs or other config details. ]

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: src/Components/OATGraphViewer/OATGraphViewer.tsx [hunk 1]
```diff
@@ -60,6 +61,8 @@ const getStoredElements = () => {
     const editorData = JSON.parse(localStorage.getItem(OATDataStorageKey));
     return editorData && editorData.models ? editorData.models : null;
 };
+const nodeWidth = 300;
+const nodeHeight = 100;
```
Comment (jamescarpinter) [inline src/Components/OATGraphViewer/OATGraphViewer.tsx:65]: Are these values present anywhere else in the code base?

File: src/Components/OATGraphViewer/OATGraphViewer.tsx [hunk 2]
```diff
@@ -90,6 +93,45 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
...
+            if (isNode(el)) {
+                const nodeWithPosition = dagreGraph.node(el.id);
+                el.targetPosition = isHorizontal ? 'left' : 'top';
+                el.sourcePosition = isHorizontal ? 'right' : 'bottom';
+
+                el.position = {
+                    x:
+                        nodeWithPosition.x -
+                        nodeWidth / 2 +
+                        Math.random() / 1000,
```
Comment (jamescarpinter) [inline src/Components/OATGraphViewer/OATGraphViewer.tsx:126]: Can you help me understand why random is required here? Non-deterministic code always gets my interest 😄

File: src/Components/OATGraphViewer/OATGraphViewer.tsx [hunk 3]
```diff
@@ -235,7 +277,8 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
                 );
                 importModelsList.push(newNode, ...relationships);
             });
-            setElements([...importModelsList]);
+            const layoutedElements = getLayoutedElements([...importModelsList]);
```
Comment (jamescarpinter) [inline src/Components/OATGraphViewer/OATGraphViewer.tsx:280]: `layoutedElements` / `getLayoutedElements` --> might be better as `positionedElements` / `applyLayoutToElements`? `layouted` isn't quite right 🤔

File: src/Components/OATModelList/OATModelList.tsx [hunk 4]
```diff
@@ -39,6 +40,17 @@ const OATModelList = ({ elements, dispatch }: OATModelListProps) => {
         setItems([...elements]);
     }, [theme]);
 
+    useEffect(() => {
+        setItems(
+            elements.filter(
+                (element) =>
+                    filter === '' ||
```
Comment (jamescarpinter) [inline src/Components/OATModelList/OATModelList.tsx:47]: Would `!filter` be more robust?

**Review by jamescarpinter — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -23,6 +23,7 @@
         "axios-retry": "^3.1.9",
         "azure-iot-dtdl-parser": "0.0.1-beta.0",
         "copy-to-clipboard": "^3.3.1",
+        "dagre": "^0.8.5",
```
[the remark this answers is not in the document]
Comment (JoseZarmada) [reply package.json:26]: Is the only example we found on the flow renderer docs, they itself don't have a layouting  option. We can explore other options similar to dagre but that will require time and research, right now we are complying with the card. 
image

File: src/Components/OATGraphViewer/OATGraphViewer.tsx (hunk 1 above)
Comment (JoseZarmada) [reply src/Components/OATGraphViewer/OATGraphViewer.tsx:65]: No those ones are average sizes for the nodes, for the floating edges we have a way to measure the exact size of each independent node, but for layouting matters that effort don't seem necessary.

File: src/Components/OATGraphViewer/OATGraphViewer.tsx (hunk 2 above)
Comment (JoseZarmada) [reply src/Components/OATGraphViewer/OATGraphViewer.tsx:126]: image
Its a little hack I found necessary for Flow renderer V9, initially was using the examples from V10 and it didn't worked, it looks they improve it and its no longer required for any V10 implementation, not our case though

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (jamescarpinter) [reply package.json:26]: @jamiewilbraham would be great to get your thoughts here.

File: src/Components/OATModelList/OATModelList.tsx [hunk 6]
```diff
@@ -39,6 +40,17 @@ const OATModelList = ({ elements, dispatch }: OATModelListProps) => {
...
+                    element['@id'].includes(filter) ||
+                    element.displayName.includes(filter)
```
[the remark this answers is not in the document]
Comment (jamescarpinter) [reply src/Components/OATModelList/OATModelList.tsx:49]: Thanks for checking 👍

File: src/Components/OATPropertyEditor/OATPropertyEditor.styles.ts [hunk 7]
```diff
@@ -401,6 +401,13 @@ export const getPropertyInspectorStyles = () => {
                 margin: '0, 4px'
             } as IStyle
         ],
+        propertyListRelativeWrapContainer: [
+            classNames.propertyListRelativeWrapContainer,
+            {
+                overflowY: 'auto',
+                maxHeight: 'calc(100vh - 310px)'
```
[the remark this answers is not in the document]
Comment (jamescarpinter) [reply src/Components/OATPropertyEditor/OATPropertyEditor.styles.ts:408]: Hmmm. Let's leave this for now. Can you please log a bug in JIRA for us to review the layout size of the components in the property grid area? I'm not sure if we have things quite right (in a bunch of places now that I've reviewed it). Thanks!

[the remark this answers is not in the document]
Comment (jamiewilbraham) [reply src/Components/OATPropertyEditor/OATPropertyEditor.styles.ts:408]: @jamescarpinter have logged a bug for this FYI

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (jamiewilbraham) [reply package.json:26]: @jamescarpinter - This was discussed today, the above is a good placeholder for a default layout. However, we will have to implement a force-directed algorithm as the default as part of the MVP. I am looking into this..

**Review by jamescarpinter — APPROVED** [review]

**Pull request merged**